ouroboros-network/src/Ouroboros/Network/Protocol/TipSample/Type.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE TypeFamilies      #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE EmptyCase         #-}
+{-# LANGUAGE PolyKinds         #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+module Ouroboros.Network.Protocol.TipSample.Type where
+
+import Network.TypedProtocol.Core
+import Network.TypedProtocol.Pipelined (N (..), Nat)
+import Cardano.Slotting.Slot (SlotNo)
+
+import Ouroboros.Network.Util.ShowProxy
+
+
+-- | There are three of blocking requests: awiat until slot, await until
+-- a tip will change; the third one has a dedicated type:
+-- @'StFollowTip' :: 'TipSample' tip@.
+--
```

Comment:
The comment doesn't match the current code. There is only one request, MsgFollowTip.
I don't think the types `TipRequestKind` and `TokTipRequest` are needed anymore.